Project/core/src/main/java/View/screen/GameScreenComponents/RevealButton.java

```diff
@@ -27,10 +25,7 @@ public RevealButton(GameController gameController, Skin skin) {
         this.addListener(new ChangeListener() {
             @Override
             public void changed(ChangeEvent event, Actor actor) {
-                // Not sure if this is working now
-                Board board = gameController.getGame().getBoard();
-                int[] playerPosition = board.getPlayerCoord(player);
-                NormalCell cell = (NormalCell) board.getCell(playerPosition[0], playerPosition[1]);
+                AbstractCell cell = gameController.getGame().getCurrentPlayerCell();
                 if (cell.containsFootstep()) {
                     gameController.actionHandler(Actions.REVEAL);
```

Comment:
This validation could probably be done in the actionHandler as actionHandler already retrieves the current cell which would remove some redundancy. I thinkMOVE does it similarly, where the handler is always run but it just runs break if it's invalid.

Reply:
Updated now!